jaxrs/src/main/java/org/jboss/as/jaxrs/ResteasyExtensionTransformerRegistration.java

```diff
@@ -32,14 +33,49 @@ public String getSubsystemName() {
     public void registerTransformers(final SubsystemTransformerRegistration subsystemRegistration) {
         ChainedTransformationDescriptionBuilder builder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(subsystemRegistration.getCurrentSubsystemVersion());
 
-        registerV3Transformers(builder.createBuilder(JaxrsExtension.CURRENT_MODEL_VERSION, VERSION_3_0_0));
+        registerV3Transformers(builder.createBuilder(VERSION_4_0_0, VERSION_3_0_0));
+        registerV4Transformers(builder.createBuilder(JaxrsExtension.CURRENT_MODEL_VERSION, VERSION_4_0_0));
 
-        builder.buildAndRegister(subsystemRegistration, new ModelVersion[] {VERSION_3_0_0, JaxrsExtension.CURRENT_MODEL_VERSION});
+        builder.buildAndRegister(subsystemRegistration, new ModelVersion[] {VERSION_3_0_0, VERSION_4_0_0, JaxrsExtension.CURRENT_MODEL_VERSION});
     }
 
     private static void registerV3Transformers(ResourceTransformationDescriptionBuilder subsystem) {
         subsystem.getAttributeBuilder()
-                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, JaxrsAttribute.TRACING_TYPE, JaxrsAttribute.TRACING_THRESHOLD)
-                .addRejectCheck(RejectAttributeChecker.DEFINED, JaxrsAttribute.TRACING_TYPE, JaxrsAttribute.TRACING_THRESHOLD);
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE,
+                        JaxrsAttribute.TRACING_TYPE, JaxrsAttribute.TRACING_THRESHOLD,
+                        JaxrsAttribute.RESTEASY_MATCH_CACHE_ENABLED,
+                        JaxrsAttribute.RESTEASY_MATCH_CACHE_SIZE,
+                        JaxrsAttribute.RESTEASY_PATCH_FILTER_DISABLED,
+                        JaxrsAttribute.RESTEASY_PATCH_FILTER_LEGACY,
+                        JaxrsAttribute.RESTEASY_ORIGINAL_WEBAPPLICATIONEXCEPTION_BEHAVIOR,
+                        JaxrsAttribute.RESTEASY_PROXY_IMPLEMENT_ALL_INTERFACES)
+                .addRejectCheck(RejectAttributeChecker.DEFINED,
+                        JaxrsAttribute.TRACING_TYPE, JaxrsAttribute.TRACING_THRESHOLD,
+                        JaxrsAttribute.RESTEASY_MATCH_CACHE_ENABLED,
+                        JaxrsAttribute.RESTEASY_MATCH_CACHE_SIZE,
+                        JaxrsAttribute.RESTEASY_PATCH_FILTER_DISABLED,
+                        JaxrsAttribute.RESTEASY_PATCH_FILTER_LEGACY,
+                        JaxrsAttribute.RESTEASY_ORIGINAL_WEBAPPLICATIONEXCEPTION_BEHAVIOR,
+                        JaxrsAttribute.RESTEASY_PROXY_IMPLEMENT_ALL_INTERFACES);
+    }
```

Comment:
This method should have have changed. Only the new method should be added.

Reply:
Ok